### DISCUSSION TITLE
Nested quotes and commas don't seem to work correctly

### DISCUSSION THREAD

**Discussion by duanbailey** [JoshClose/CsvHelper#1843] [Q&A] [answered]
Hi there!

I'm trying to read a csv file with nested quotes and commas, but it doesn't parse it as I expect.

So my csv looks like this:

```
customer_name,id
""The Literal", My Fake Company Limited",123456
My Other Fake Company, 98765
```

When I run this using 

```
var config = new CsvConfiguration(CultureInfo.InvariantCulture)
{
	HasHeaderRecord = true,
	BadDataFound = null,
};

using (var streamReader = new StreamReader(_csvFile))
{
	using (var reader = new CsvReader(streamReader, config))
	{
		var records = reader.GetRecords<dynamic>().ToList()
	}
}
```

I get "The Literal" and "My Fake Comany Limited" as the first two columns, though I would expect to see "The Literal", My Fake Company Limited and 123456.

Any ideas?

**Comment by AltruCoder** [comment, marked as answer]
Your internal quotes need to be escaped with another double quote.
```
customer_name,id
"""The Literal"", My Fake Company Limited",123456
My Other Fake Company, 98765
```

**Reply by Quackers2446** [reply]
Sadly I believe the quotes are auto-generated when taken from a spreadsheet, as the original value is: 

`"The Literal", My Fake Company Limited`

Is there any other way to do this?

**Reply by duanbailey** [reply]
> Your internal quotes need to be escaped with another double quote.
> 
> ```
> customer_name,id
> """The Literal"", My Fake Company Limited",123456
> My Other Fake Company, 98765
> ```

Thanks for the reply! So if I don't control the input and the csv's are provided by the customer is the only way to get around this to do some preprocessing? If yes, is there a best practice on this? Apologies, I'm fairly new to CsvHelper.

**Reply by JoshClose** [reply]
What happens if there is a line break or or comma in a field? Does the file you get from the customer handle that at all?

**Reply by duanbailey** [reply]
So in this case, there are quotes that wrap the comma, so the customer is expecting them to show up in the field. Without the quotes, the comma would just be read as a delimiter (which is what it does now). I believe in the case of a line break, I expect that would be considered invalid data.

**Reply by JoshClose** [reply]
Quotes wrap just the comma or the whole field?

**Reply by duanbailey** [reply]
Sorry. The whole field. So for example, it might look like:

```
""The Literal", My Fake Company Limited",123456
```

Edit: So to be clear, there are two fields here, `"The Literal", My Fake Company Limited` and `123456`.

**Reply by AltruCoder** [reply]
Which spreadsheet program is your customer using?  If I save an Excel sheet as CSV, the field `"The Literal", My Fake Company Limited` is converted to `"""The Literal"", My Fake Company Limited"`.  It also works that way with Google Sheets.  Open Office works that way if you select ISO-8859-1 for the output.  If you select UTF-8, Open Office converts the inner quotes to `“` and `”`, which I don't like, but will still work in CsvHelper.

**Reply by AltruCoder** [reply]
I think you are going to have difficulty even preprocessing the data.  I'm assuming the files you are getting will be more complicated than your example text.  There would be a bit of logic involved to process the following text and I'm pretty sure there would be more than one way to split the text into the 4 columns no matter what logic you tried to give it.
```
customer_name,customer_branch,customer_division,id
""The Literal", My Fake,", Company Limited",Branch 1,","Division" has "A Literal",As Well",123456
```

**Reply by duanbailey** [reply]
@AltruCoder we went back and checked the original data and it was indeed triple quoted. So I think there was a mix up on our end somewhere. When we update to use version 25.0 of the libraries we do seem to be parsing correctly so thanks very much for your help!
